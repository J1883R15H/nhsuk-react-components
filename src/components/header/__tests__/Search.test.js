import React from 'react';
import { shallow, mount } from 'enzyme';
import Search from '../components/search';

describe('Header.Search', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Search></Search>);
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
  it('opens the search', () => {
    const wrapper = mount(<Search></Search>);
    expect(wrapper.find('.nhsuk-header__search-wrap').props().className).toBe(
      'nhsuk-header__search-wrap'
    );
    wrapper.find('#toggle-search').simulate('click');
    expect(wrapper.find('.nhsuk-header__search-wrap').props().className).toBe(
      'nhsuk-header__search-wrap js-show'
    );
    wrapper.find('#close-search').simulate('click');
    expect(wrapper.find('.nhsuk-header__search-wrap').props().className).toBe(
      'nhsuk-header__search-wrap'
    );
    wrapper.unmount();
  });
  it('has good defaultProps', () => {
    expect(Search.defaultProps.onSubmit()).toBe(undefined);
  });
});
