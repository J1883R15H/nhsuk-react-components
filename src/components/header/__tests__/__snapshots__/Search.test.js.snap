// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.Search matches snapshot 1`] = `
<div
  className="nhsuk-header__search"
>
  <button
    aria-controls="search"
    aria-label="Open search"
    className="nhsuk-header__search-toggle"
    id="toggle-search"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="nhsuk-icon nhsuk-icon__search"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"
      />
    </svg>
    <span
      className="nhsuk-u-visually-hidden"
    >
      Search
    </span>
  </button>
  <div
    className="nhsuk-header__search-wrap"
  >
    <form
      action=""
      className="nhsuk-header__search-form"
      method=""
      onSubmit={[Function]}
      role="search"
    >
      <label
        className="nhsuk-u-visually-hidden"
        htmlFor=""
      >
        Search the NHS website
      </label>
      <div
        className="autocomplete-container"
        id="autocomplete-container"
      />
      <input
        autoComplete="off"
        className="nhsuk-search__input"
        id=""
        name="search-field"
        placeholder="Search"
        type="search"
      />
      <button
        className="nhsuk-search__submit"
        type="submit"
      >
        <svg
          aria-hidden="true"
          className="nhsuk-icon nhsuk-icon__search"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19.71 18.29l-4.11-4.1a7 7 0 1 0-1.41 1.41l4.1 4.11a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 10a5 5 0 1 1 5 5 5 5 0 0 1-5-5z"
          />
        </svg>
        <span
          className="nhsuk-u-visually-hidden"
        >
          Search
        </span>
      </button>
      <button
        className="nhsuk-search__close"
        id="close-search"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="nhsuk-icon nhsuk-icon__close"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13.41 12l5.3-5.29a1 1 0 1 0-1.42-1.42L12 10.59l-5.29-5.3a1 1 0 0 0-1.42 1.42l5.3 5.29-5.3 5.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0l5.29-5.3 5.29 5.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
          />
        </svg>
        <span
          className="nhsuk-u-visually-hidden"
        >
          Close search
        </span>
      </button>
    </form>
  </div>
</div>
`;
